WordPress/Sniffs/WP/I18nSniff.php

```diff
@@ -392,4 +407,118 @@ protected function check_text( PHP_CodeSniffer_File $phpcs_file, $context ) {
 		}
 	} // End check_text().
 
+	/**
+	 * Check for the presence of a translators comment if one of the text strings contains a placeholder.
+	 *
+	 * @param PHP_CodeSniffer_File $phpcs_file The file being scanned.
+	 * @param int                  $stack_ptr  The position of the gettext call token
+	 *                                         in the stack.
+	 * @param array                $args       The function arguments.
+	 * @return void
+	 */
+	protected function check_for_translator_comment( PHP_CodeSniffer_File $phpcs_file, $stack_ptr, $args ) {
+		$tokens = $phpcs_file->getTokens();
+
+		foreach ( $args as $arg ) {
+			if ( false === in_array( $arg['arg_name'], array( 'text', 'single', 'plural' ), true ) ) {
+				continue;
+			}
+
+			foreach ( $arg['tokens'] as $token ) {
+				if ( empty( $token['content'] ) ) {
+					continue;
+				}
+
+				if ( preg_match( self::SPRINTF_PLACEHOLDER_REGEX, $token['content'], $placeholders ) < 1 ) {
+					// No placeholders found.
+					continue;
+				}
+
+				$previous_comment = $phpcs_file->findPrevious( PHP_CodeSniffer_Tokens::$commentTokens, ( $stack_ptr - 1 ) );
+
+				if ( false !== $previous_comment ) {
+					/*
+					 * Check that the comment is either on the line before the gettext call or
+					 * if it's not, that there is only whitespace between.
+					 */
+					$correctly_placed = false;
+
+					if ( ( $tokens[ $previous_comment ]['line'] + 1 ) === $tokens[ $stack_ptr ]['line'] ) {
+						$correctly_placed = true;
+					} else {
+						$next_non_whitespace = $phpcs_file->findNext( T_WHITESPACE, ( $previous_comment + 1 ), $stack_ptr, true );
+						if ( false === $next_non_whitespace || $tokens[ $next_non_whitespace ]['line'] === $tokens[ $stack_ptr ]['line'] ) {
+							// No non-whitespace found or next non-whitespace is on same line as gettext call.
+							$correctly_placed = true;
+						}
+						unset( $next_non_whitespace );
+					}
+
+					/*
+					 * Check that the comment starts with 'translators:'.
+					 */
+					if ( true === $correctly_placed ) {
+
+						if ( T_COMMENT === $tokens[ $previous_comment ]['code'] ) {
+							$comment_text = trim( $tokens[ $previous_comment ]['content'] );
+
+			  		   		// If it's multi-line /* */ comment, collect all the parts.
+			  		   		if ( '*/' === substr( $comment_text, -2 ) && '/*' !== substr( $comment_text, 0, 2 ) ) {
```

Comment:
Could this "collect all parts from a multi-line comment" be abstracted into a re-usable method in the base class?

How about :

```
/*
 * This is a multiline comment,
 * But it also has * at the start
 of some lines ;-)
*/
```